event_systemhook_types.go

```diff
@@ -130,6 +130,41 @@ type UserTeamSystemEvent struct {
 // https://docs.gitlab.com/ee/system_hooks/system_hooks.html
 type PushSystemEvent struct {
 	BaseSystemEvent
+	Before       string `json:"before"`
+	After        string `json:"after"`
+	Ref          string `json:"ref"`
+	CheckoutSha  string `json:"checkout_sha"`
+	UserID       int    `json:"user_id"`
+	UserName     string `json:"user_name"`
+	UserUsername string `json:"user_username"`
+	UserEmail    string `json:"user_email"`
+	UserAvatar   string `json:"user_avatar"`
+	ProjectID    int    `json:"project_id"`
+	Project      struct {
+		Name              string `json:"name"`
+		Description       string `json:"description"`
+		WebUrl            string `json:"web_url"`
+		AvatarUrl         string `json:"avatar_url"`
+		GitHttpUrl        string `json:"git_http_url"`
+		GitSshUrl         string `json:"git_ssh_url"`
+		Namespace         string `json:"namespace"`
+		VisibilityLevel   int    `json:"visibility_level"`
+		PathWithNamespace string `json:"path_with_namespace"`
+		DefaultBranch     string `json:"default_branch"`
+		Homepage          string `json:"homepage"`
+		Url               string `json:"url"`
+	} `json:"project"`
+	Commits []struct {
```

Comment:
Can we use the `Commit` struct instead? Or is the returned data too different?

Reply:
If you are refering to the `Commit` struct from [commits.go:37](https://github.com/xanzy/go-gitlab/blob/3ac36fa074a445904aa841c77fb58973ffdb3648/commits.go#L37), then yes it is too different.

FYI most of the structs in `event_webhook_types.go` are also using anonymous structs as they almost always differ.
Eg. https://github.com/xanzy/go-gitlab/blob/3ac36fa074a445904aa841c77fb58973ffdb3648/event_webhook_types.go#L47-L57
https://github.com/xanzy/go-gitlab/blob/3ac36fa074a445904aa841c77fb58973ffdb3648/event_webhook_types.go#L112-L122
https://github.com/xanzy/go-gitlab/blob/3ac36fa074a445904aa841c77fb58973ffdb3648/event_webhook_types.go#L816-L826